Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,523 @@
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableAADForKubernetesServices"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+
+    if (-not $DryRun)
+    {
+        $colsProperty = @{Expression={$_.ResourceId};Label="Resource ID";Width=40;Alignment="left"},
+                        @{Expression={$_.ResourceGroupName};Label="Resource Group Name";Width=20;Alignment="left"},
+                        @{Expression={$_.ResourceName};Label="Resource Name";Width=20;Alignment="left"},
+                        @{Expression={$_.Location};Label="Location";Width=20;Alignment="left"},
+                        @{Expression={$_.isRBACEnabled};Label="Is RBAC enabled on Kubernetes Cluster?";Width=20;Alignment="left"},
+                        @{Expression={$_.isAADEnabled};Label="Is AAD enabled on Kubernetes Cluster?";Width=20;Alignment="left"}
+
+        $kubernetesServicesWithoutAADEnabled | Format-Table -Property $colsProperty -Wrap
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "`n[Step 3 of 4] Backing up Kubernetes Service details to $($backupFolderPath)"
+        
+        $backupFile = "$($backupFolderPath)\KubernetesClusterWithAADDisabled.csv"
+        $kubernetesServicesWithoutAADEnabled | Export-CSV -Path $backupFile -NoTypeInformation
+        
+        Write-Host "Kubernetes Service details have been backed up to $($backupFile)" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host "AAD will be enabled for all Kubernetes Services."
+        Write-Host "Do you want to add Azure AD groups as administrators on each cluster?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+        
+        $addAADGroup = $false
+        $input = Read-Host -Prompt "(Y|N)"
+
+        if($input -eq "Y")
+        {
+            $addAADGroup = $true
+        }
+
+        if (-not $Force)
+        {
+            Write-Host "Once AAD is enabled, you won't be able to disable again." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host "Do you want to enable AAD for all Kubernetes Services? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "AAD will not be enabled for any Kubernetes Services. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+                break
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. AAD will be enabled for all Kubernetes Services without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
Force switch can be riskly here.

Reply:
Removed Force switch.
There would be user interaction while running the script to avoid risk.